### PR TITLE
T17240

### PR DESCRIPTION
**Pull request by smspillaz**
Bring back coverage reporting.

In order to get this to work, we needed to carry forward patches to port the tests to jasmine and refactor some of the shell code so as to avoid cross-dependencies. Some of this work is a good candidate for upstreaming.

https://phabricator.endlessm.com/T17240

### CODE REVIEW

File: js/misc/messageListMarkup.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// -*- mode: js; js-indent-level: 4; indent-tabs-mode: nil -*-
+
+const GLib = imports.gi.GLib;
+const Pango = imports.gi.Pango;
+
+function fixMarkup(text, allowMarkup) {
+    log("Should allow markup " + allowMarkup);
```
Comment (cosimoc) [inline js/misc/messageListMarkup.js]: Remove debug leftover

File: js/misc/coreEnvironment.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+function coreInit() {
+    window.log = _makeLoggingFunc(window.log);
+
+    window._ = Gettext.gettext;
+    window.C_ = Gettext.pgettext;
+    window.ngettext = Gettext.ngettext;
+    window.N_ = function(s) { return s; };
+
+    const Format = imports.format;
+    String.prototype.format = Format.format;
```
Comment (cosimoc) [inline js/misc/coreEnvironment.js]: Should you not remove these from the non-core environment then?

File: js/ui/sessionMode.js [hunk 3]
```diff
@@ -106,8 +106,8 @@ const _modes = {
         panel: {
             left: ['endlessButton', 'appIcons'],
             center: [],
-            right: ['a11y', 'keyboard', 'aggregateMenu', 'userMenu',
-                    'dateMenu', 'hotCorner']
+            right: ['dateMenu', 'a11y', 'keyboard', 'aggregateMenu', 'userMenu',
```
Comment (cosimoc) [inline js/ui/sessionMode.js]: I don't think this change is right. The clock should stay where it currently is.

Comment (smspillaz) [inline js/ui/sessionMode.js]: Yep, @rshuler confirmed that too. I'll update the test, in that case.

File: js/misc/coreEnvironment.js (hunk 2 above)
Comment (smspillaz) [inline js/misc/coreEnvironment.js]: I think they should be removed ... unless I missed them.

**Comment by smspillaz** [comment]
Comments addressed. As for the test naming - I did try and revert that commit, but it looks like it breaks Jasmine. If I remember correctly, Jasmine required 'test' to be in the name of the file before it would execute it.

File: js/misc/messageListMarkup.js (hunk 1 above)
Comment (cosimoc) [inline js/misc/messageListMarkup.js]: Still should be removed

**Comment by smspillaz** [comment]
> Still should be removed

Oof, I thought I removed that :( I'll remove it now.

**Comment by smspillaz** [comment]
Upstream bug: https://bugzilla.gnome.org/show_bug.cgi?id=783738

**Comment by smspillaz** [comment]
There's a bunch of comments on the upstream patchset which I'd like to fix and port back downstream before merging this, see the linked bug report.

**Comment by mariospr** [comment]
I'm closing this for now, as it's been open since June and does not really apply anymore.

@smspillaz if you want to rebase it and work on it after the upgrade to 3.26, then it's fine but for now it's just cluttering the list of open PRs.

**Pull request closed without merging**